src/storage/src/source/kafka.rs

```diff
@@ -891,7 +895,11 @@ impl KafkaSourceReader {
         self.create_partition_queue(pid, Offset::Offset(start_offset));
 
         let prev = self.last_offsets.insert(pid, start_offset - 1);
+        assert!(prev.is_none());
 
+        let prev = self
+            .partition_as_of
+            .insert(pid, watermarks.high.saturating_sub(1));
```

Comment:
I may be missing something, but how is this re-populated after a restart but once this partition has some offsets committed? Doesn't it reset to the high watermark offset each time the operator starts?